pkg/resource/resid.go

```diff
@@ -28,18 +28,36 @@ type ResId struct {
 	gvk schema.GroupVersionKind
 	// original name of the resource before transformation.
 	name string
+	// namePrefix of the resource
+	prefix string
 }
 
 // NewResId creates new resource identifier
-func NewResId(g schema.GroupVersionKind, n string) ResId {
-	return ResId{gvk: g, name: n}
+func NewResId(g schema.GroupVersionKind, n ...string) ResId {
```

Comment:
no, this trick uses a string array to pass two args with different semantics, and silently discards longer arrays.

Prefer a new method, e.g.
```
 NewResIdWithPrefix(s, n, p)
```
The original NewResId would be changed to call it, i.e.
```
func NewResId(g, n ) ResId { return NewResIdWithPrefix(g, n, "") }
```
and there's no need to update existing call points.